package.json

```diff
@@ -63,7 +63,7 @@
     "typescript": "^4.2.3"
   },
   "engines": {
-    "node": ">=12.14.1"
+    "node": ">=14.15.0"
```

Comment:
If we don't enforce individual packages, should we enforce the top level? I suppose it's never `npm install`ed anywhere directly, but wondering what the behavior of `yarn install` is.

Reply:
Doubled checked in a container running node 12 and `yarn install` fails, so this change should move to the second commit (or be reverted if we decide not to include the second commit)

Reply:
I think we want `yarn install` to fail if the user's Node is too old (fail fast), so I'm pretty sure we want the top-level enforcement.

The purpose of the bottom-level `package.json` changes would be to make a similar enforcement on individual packages if e.g. someone installed one from NPM, rather than using our monorepo. I know we want this for at least swingset (where I'll be making a 14.x-requiring change soon). What I don't know is if it's better to have all the packages list the same requirement (yay consistency), or have each only list the requirement we know is true (yay tolerance of individual packages being used via NPM on older systems).

If we choose the former, I'll include the second commit. If we choose the latter, I'd change that commit to only update swingset (or, as @michaelfig suggested, move that `package.json` change into my upcoming PR that actually depends on WeakRef). I don't want to exhaustively test each package against different versions to see what it really requires, so I'd simply assume that the current individual requirements are suffciently correct and let them stand.

So I guess my real question is: is consistency of the requirements important/useful?

Reply:
Thanks for the detailed reasoning.

Given that we currently publish individual packages to NPM, I think consistency is not as valuable. We can enforce consistency for development by keeping the root `package.json` to the latest engine version required by any of the required sub packages.

Technically that would mean dropping the 2nd commit and only requiring whatever latest is in this PR, before bumping both the SwingSet and root package.json in the next PR.

Reply:
> Technically that would mean dropping the 2nd commit and only requiring whatever latest is in this PR, before bumping both the SwingSet and root package.json in the next PR.

Ok, are we comfortable with a commit which drops CI coverage for 12, but leaves the top-level package.json claiming `node >=12.14.1`?

Reply:
Yeah I guess we should ignore the PR layering here and just require 14 for the top level.